operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfiguration.java

```diff
@@ -17,7 +17,15 @@ default String getName() {
   }
 
   default String getResourceTypeName() {
-    return HasMetadata.getFullResourceName(getResourceClass());
+    return getResourceTypeName(getResourceClass());
+  }
+
+  static <T extends HasMetadata> String getResourceTypeName(Class<T> resourceClass) {
```

Comment:
Same here, pls add unit test.

Reply:
This is a workaround for a fix that will be in fabric8 5.12 where it's already pretty well tested.

Reply:
But if too much effort, I'm ok to skip it.